R/dataset_template.R

```diff
@@ -0,0 +1,116 @@
+#' DatasetTemplate.all
+#'
+#' Retrieves the metadata about all dataset templates on SolveBio.
+#'
+#' @param env (optional) Custom client environment.
+#' @param ... (optional) Additional query parameters (e.g. page).
+#'
+#' @examples \dontrun{
+#' DatasetTemplate.all()
+#' }
+#'
+#' @references
+#' \url{https://docs.solvebio.com/}
+#'
+#' @export
+DatasetTemplate.all <- function(env = solvebio:::.solveEnv, ...) {
+    .request('GET', "v2/dataset_templates", query=list(...), env=env)
+}
```

Comment:
can't seem to get the is_recipe flag to work, but maybe it's my ignoRance
```
> DatasetTemplate.all()$total
[1] 2326
> DatasetTemplate.all(tags="import")$total
[1] 7
> DatasetTemplate.all(is_recipe=TRUE)$total
[1] 2326
```

Reply:
`DatasetTemplate.all(is_recipe="True")$total`

This is an API issue (filed as a separate ticket).